### PR TITLE
SWATCH-2744: Fix capacity reconciliation when another task is queued

### PR DESCRIPTION
**Pull request by kahowell**
https://issues.redhat.com/browse/SWATCH-2744

See inline comments for more details.

Testing
-------

See `CapacityReconciliationIntegrationTest`. To reproduce the issue, revert the fix and run the test.

### CODE REVIEW

**Review by kflahert — APPROVED** [review]

**Comment by kflahert** [comment]
LGTM, looking into that weird testcontainers error.

**Comment by kahowell** [comment]
5472eab to e0be4b4: address sonarcloud issues

**Review by ntkathole — APPROVED** [review]
Looks good!

File: swatch-contracts/src/main/java/com/redhat/swatch/contract/service/CapacityReconciliationService.java [hunk 1]
```diff
@@ -78,8 +78,18 @@ public void reconcileCapacityForOffering(String sku, int offset, int limit) {
     subscriptions.forEach(this::reconcileCapacityForSubscription);
     if (subscriptions.size() >= limit) {
       offset = offset + limit;
-      reconcileCapacityByOfferingEmitter.send(
-          ReconcileCapacityByOfferingTask.builder().sku(sku).offset(offset).limit(limit).build());
+      reconcileCapacityByOfferingEmitter
+          .send(
```
Comment (Sgitario) [inline swatch-contracts/src/main/java/com/redhat/swatch/contract/service/CapacityReconciliationService.java:82]: There is an easier way to do this by changing the `Emitter` to use `MutinyEmitter` which has the method `sendAndAwait` that does the wait already. 
This is just a minor suggestion, tho we also use the `MutinyEmitter` in other producers in our project.

File: swatch-contracts/src/test/java/com/redhat/swatch/contract/CapacityReconciliationIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+@Disabled("Running kafka w/ kubedock broken, see https://github.com/joyrex2001/kubedock/issues/53")
+@QuarkusTest
+@QuarkusTestResource(value = KafkaCompanionResource.class, restrictToAnnotatedClass = true)
+class CapacityReconciliationIntegrationTest {
+  @InjectKafkaCompanion KafkaCompanion companion;
```
Comment (Sgitario) [inline swatch-contracts/src/test/java/com/redhat/swatch/contract/CapacityReconciliationIntegrationTest.java:49]: I know this issue cannot be reproduced using the smallrye in-memory component, so I like the idea of keeping this test here, but disabled, just to troubleshooting purposes.

**Review by Sgitario — COMMENTED** [review]
@kahowell just a minor suggestion

**Comment by kahowell** [comment]
Will merge after CI passes since the suggested change I pushed was semantically equivalent.

**Pull request merged**